core/sqf/src/seatrans/hbase-trx/src/main/java/org/apache/hadoop/hbase/coprocessor/transactional/TrxRegionEndpoint.java.tmpl

```diff
@@ -7335,6 +7338,245 @@ TrxTransactionState isTransactionOnCommittedList(final long transactionId)
+    // size in bytes stored in the trailer block.
+    
+    // For aligned format tables, the number of rows equals the
+    // number of KeyValue entries. For non-aligned format, it's
+    // more complicated. There is a KeyValue entry for each 
+    // column value, except the KeyValue may be missing because
+    // the column has a null value or because the column has a
+    // default value that has not been materialized.
+
+    // For non-aligned format tables, we sample some rows and
+    // count how many entries there are per row, so our caller
+    // can estimate the average number of missing values per row.
+    // Once our caller has that estimate, it can estimate the
+    // number of rows.
+
+    // We only do the sampling for non-aligned tables (numCols > 1),
+    // and we only do it on the first HFile of the first Region.
+    // The first Region is detected by having a null start key.
+
+    CacheConfig cacheConf = new CacheConfig(config);
+    byte[] startKey = regionInfo.getStartKey();
+
+    // Get the list of store files in column family '#1'. There might
+    // not be any. For example, a new Trafodion table might be entirely
+    // in memstore with nothing written out yet. Or we may be accessing
+    // a native HBase table which lacks the '#1' colum family.
+    List<String> storeFileList = null;
+    try {
+      byte[] familyName = "#1".getBytes();
+      byte[][] familyNames = { familyName };
+      storeFileList = m_Region.getStoreFileList(familyNames);
+    }
+    catch (IllegalArgumentException iae) {
+      // this gets thrown when the column family doesn't exist;
+      // we'll just use an empty list instead
+      storeFileList = new ArrayList<String>();
+    }
+
+    if (LOG.isDebugEnabled()) {
+      LOG.debug("Trafodion estimate row count sees " + storeFileList.size() + " files.");
+      for (String sfn : storeFileList) {
+        LOG.debug("*** " + sfn);
+      }
+      if (startKey == null)
+        LOG.debug("startKey is null.");
+      else
+        LOG.debug("startKey.length is " + startKey.length + ", startKey is hex " + bytesToHex(startKey));
+    }
+ 
+    try {
+      FileSystem fileSystem = FileSystem.get(config);
+
+      for (String storeFileName : storeFileList) {
+        Path path = new Path(storeFileName);
+        // Make sure the file name conforms to HFile name pattern.
+        if (!StoreFileInfo.isHFile(path)) {
+          if (LOG.isDebugEnabled()) LOG.debug("Trafodion estimate row count file name " + path + " is not an HFile.");
+          continue;
+        }
+        HFile.Reader reader = HFile.createReader(fileSystem, path, cacheConf, config);
+        try {
+          totalEntries += reader.getEntries();
+          totalSizeBytes += reader.length();        
+          if (ROWS_TO_SAMPLE > 0 &&
+              numCols > 1 &&  // only need to do this for non-aligned format
+              (startKey == null || startKey.length == 0) &&  // first region only
```

Comment:
Minor nit perhaps, but startKey should never be null here.  If it is, that would be some sort of error, but you're allowing it.  I think it would be better to test for the HConstants.EMPTY_START_ROW, which is different than a null, or remove the test for null and just your test of length of 0 would also be sufficient.